Firestore/core/src/firebase/firestore/local/leveldb_transaction.cc

```diff
@@ -164,7 +161,7 @@ const WriteOptions& LevelDbTransaction::DefaultWriteOptions() {
 
 void LevelDbTransaction::Put(std::string key, std::string value) {
   deletions_.erase(key);
-  mutations_.emplace(std::make_pair(std::move(key), std::move(value)));
+  mutations_[std::move(key)] = std::move(value);
```

Comment:
So I think this was a bug that didn't manifest because the Objective-C overload of this function stores the value in the map in a different way, using `operator[]`; I noticed it because the `nanopb::Message` overload of `Put` that I added delegates to this implementation.

The issue is that `operator[]` overwrites an existing key, while `emplace` doesn't (instead, it returns a value indicating whether an element was added). 8 tests in our suite started failing once this version of `Put` started being used exclusively (either directly or through the `Message` overload I added).

Reply:
Good catch!